src/prof_recent.c

```diff
@@ -15,11 +15,19 @@
 #endif
 
 ssize_t opt_prof_recent_alloc_max = PROF_RECENT_ALLOC_MAX_DEFAULT;
-malloc_mutex_t prof_recent_alloc_mtx; /* Protects the fields below */
+
+malloc_mutex_t prof_recent_alloc_mtx; /* Protects the fields below. */
 static atomic_zd_t prof_recent_alloc_max;
 static ssize_t prof_recent_alloc_count = 0;
 static prof_recent_t *prof_recent_alloc_dummy = NULL;
 
+malloc_mutex_t prof_recent_dump_mtx; /* Protects dumping & the fields below. */
+/*
+ * prof_recent_alloc_dummy_reserve points to an empty list to be used during
+ * dumping, so that there's no need to do allocate / free there.
+ */
+static prof_recent_t *prof_recent_alloc_dummy_reserve = NULL;
```

Comment:
I think this, and prof_recent_alloc_dummy above, could still use some more comments. This explains where it's used, but not what it is.

E.g.
- Who puts things into the list, and who takes them out?
- Why do those two things need to communicate through a global, instead of just one being the caller and one the callee?
- Why do we need both prof_recent_alloc_dummy and prof_recent_alloc_dummy_reserve? What's the difference between them?

Reply:
Added more comments. Actually, the more I think of it, the more I feel that a global `prof_recent_alloc_dummy_reserve` is not necessary. The comment I'm writing sounds defensive, where I try to enumerate all downsides of not defining it global, but I'm personally neutral now: defining it global has the design downside, and I'm not sure how that compares with the downsides I listed.

Reply:
Leaning slightly towards keeping `prof_recent_alloc_dummy_reserve` as a global variable, mainly for implementation simplicity. Thoughts?

Reply:
Ahh, I think I understand why the dummy node (/ reserve) here. This is really just part of a linked-list maintenance strategy IIUC.

I think probably, instead of hand-rolling the list-manipulation code, it would probably be better to use the ql module (our linked list code), which doesn't have a need for dummy nodes. (Possibly extended with a "pop all" functionality or something"). This would

I think this is probably more along the lines of what Qi meant down below -- rather than assigning through a `prof_recent_t **`, we can work in terms of a `prof_recent_list_t *to_delete` and just append onto it.